output error while running the chain-bench scan

## Description

Executing this simple command: _docker run aquasec/chain-bench scan --repository-url [git repo] --access-token [git token]_

Produce these errors:
2022-07-28 03:44:50 INF 🚩      Fetch Starting
2022-07-28 03:44:50 ERR error in authenticated user data
2022-07-28 03:44:50 ERR error in fetching repository data
2022-07-28 03:44:50 INF 🛢️       Fetching Repository Settings Finished
2022-07-28 03:44:50 ERR error in fetching branch protection
2022-07-28 03:44:50 INF 🌱      Fetching Branch Protection Settings Finished
2022-07-28 03:44:50 ERR error in fetching workflows
2022-07-28 03:44:50 INF 🔧      Fetching Pipelines Finished
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xb3f265]

goroutine 1 [running]:
github.com/aquasecurity/chain-bench/internal/scm-clients/clients.FetchClientData({0x0, 0x0}, {0x7fffb403df52?, 0x1?})
        /home/runner/work/chain-bench/chain-bench/internal/scm-clients/clients/clients.go:48 +0x3c5
github.com/aquasecurity/chain-bench/internal/commands.NewScanCommand.func1(0xc000242280?, {0xcfc484?, 0x2?, 0x2?})
        /home/runner/work/chain-bench/chain-bench/internal/commands/scan.go:22 +0xac
github.com/spf13/cobra.(*Command).execute(0xc000242280, {0xc00024c040, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000242000)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
github.com/aquasecurity/chain-bench/internal/commands.Execute({0xe4c9a8?, 0xc0000021a0?})
        /home/runner/work/chain-bench/chain-bench/internal/commands/root.go:21 +0x32
main.main()
        /home/runner/work/chain-bench/chain-bench/cmd/chain-bench/main.go:12 +0x27

Same Issue for me also. I don't see any issue on token permission or the repository. Looking forward if someone fix this similar issue. 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1741885]

goroutine 1 [running]:
github.com/aquasecurity/chain-bench/internal/scm-clients/clients.FetchClientData({0x7ffeefbffafb, 0x28}, {0x7ffeefbffaba?, 0x1?})
	github.com/aquasecurity/chain-bench/internal/scm-clients/clients/clients.go:48 +0x3c5
github.com/aquasecurity/chain-bench/internal/commands.NewScanCommand.func1(0xc000264280?, {0x1900f9c?, 0x4?, 0x4?})

Regards,
Venkat

Got the same panic when authentication token was not provided.
The repository function return `nil` object to `repo` which isn't validated and tried to be accessed via `*repo.Owner.Type`

```bash
➜  ~ docker run aquasec/chain-bench scan --repository-url https://github.com/aquasecurity/chain-bench -v
2022-09-14 13:31:33 INF 🚩	Fetch Starting
2022-09-14 13:31:33 ERR error in authenticated user data
2022-09-14 13:31:33 DBG error in authenticated user data error="GET https://api.github.com/user: 401 Bad credentials []"
2022-09-14 13:31:34 ERR error in fetching repository data
2022-09-14 13:31:34 DBG error in fetching repository data error="GET https://api.github.com/repos/aquasecurity/chain-bench: 401 Bad credentials []"
2022-09-14 13:31:34 INF 🛢️	Fetching Repository Settings Finished
2022-09-14 13:31:34 ERR error in fetching branch protection
2022-09-14 13:31:34 DBG error in fetching branch protection error="GET https://api.github.com/repos/aquasecurity/chain-bench/branches//protection: 401 Bad credentials []"
2022-09-14 13:31:34 INF 🌱	Fetching Branch Protection Settings Finished
2022-09-14 13:31:34 ERR error in fetching workflows
2022-09-14 13:31:34 DBG error in fetching workflows error="GET https://api.github.com/repos/aquasecurity/chain-bench/actions/workflows: 401 Bad credentials []"
2022-09-14 13:31:34 INF 🔧	Fetching Pipelines Finished
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xb3f265]

goroutine 1 [running]:
github.com/aquasecurity/chain-bench/internal/scm-clients/clients.FetchClientData({0x0, 0x0}, {0x7ffeb8378f4b?, 0x1?})
	/home/runner/work/chain-bench/chain-bench/internal/scm-clients/clients/clients.go:48 +0x3c5
github.com/aquasecurity/chain-bench/internal/commands.NewScanCommand.func1(0xc000250280?, {0xcfc484?, 0x3?, 0x3?})
	/home/runner/work/chain-bench/chain-bench/internal/commands/scan.go:22 +0xac
github.com/spf13/cobra.(*Command).execute(0xc000250280, {0xc0001fdf20, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000250000)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
github.com/aquasecurity/chain-bench/internal/commands.Execute({0xe4c9a8?, 0xc0000021a0?})
	/home/runner/work/chain-bench/chain-bench/internal/commands/root.go:21 +0x32
main.main()
	/home/runner/work/chain-bench/chain-bench/cmd/chain-bench/main.go:12 +0x27
```

--- a/internal/scm-clients/clients/clients.go
+++ b/internal/scm-clients/clients/clients.go
@@ -103,8 +103,8 @@ func getRepoInfo(repoFullUrl string) (string, string, string, error) {
 		return "", "", "", fmt.Errorf("missing org/repo in the repository url: %s", repoFullUrl)
 	}
 	repo := path[len(path)-1]
-	namespace := strings.Split(u.Path, repo)[0]
-	trimedNamespace := namespace[1:(len(namespace) - 1)]
+	namespace := strings.Join(path[:len(path)-1], "/")
+	trimedNamespace := strings.TrimLeft(namespace, "/")
 
 	return u.Host, trimedNamespace, repo, nil
 }

--- a/internal/scm-clients/clients/clients_test.go
+++ b/internal/scm-clients/clients/clients_test.go
@@ -54,7 +54,7 @@ func TestGetRepoInfo(t *testing.T) {
 		Name:        "github project under sub org with same name as repo",
 		RepoUrl:     "https://github.com/codekuu/suborg/secondsuborg/secondsuborg",
 		ExpectedErr: nil,
-		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},
+		Expected:    RepoInfo{BaseUrl: "github.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},
 	}}
 
 	for _, test := range tests {

--- a/internal/scm-clients/clients/clients_test.go
+++ b/internal/scm-clients/clients/clients_test.go
@@ -52,7 +52,7 @@ func TestGetRepoInfo(t *testing.T) {
 		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "rootgroup/subgroup/secondsubgroup", Project: "secondsubgroup"},
 	}, {
 		Name:        "github project under sub org with same name as repo",
-		RepoUrl:     "https://github.com/codekuu/suborg/secondsuborg/secondsuborg",
+		RepoUrl:     "https://gitlab.com/codekuu/suborg/secondsuborg/secondsuborg",
 		ExpectedErr: nil,
 		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},
 	}}

--- a/internal/scm-clients/clients/clients_test.go
+++ b/internal/scm-clients/clients/clients_test.go
@@ -54,7 +54,7 @@ func TestGetRepoInfo(t *testing.T) {
 		Name:        "github project under sub org with same name as repo",
 		RepoUrl:     "https://github.com/codekuu/suborg/secondsuborg/secondsuborg",
 		ExpectedErr: nil,
-		Expected:    RepoInfo{BaseUrl: "github.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},
+		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},
 	}}
 
 	for _, test := range tests {

--- a/internal/scm-clients/clients/clients_test.go
+++ b/internal/scm-clients/clients/clients_test.go
@@ -51,7 +51,7 @@ func TestGetRepoInfo(t *testing.T) {
 		ExpectedErr: nil,
 		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "rootgroup/subgroup/secondsubgroup", Project: "secondsubgroup"},
 	}, {
-		Name:        "github project under sub org with same name as repo",
+		Name:        "gitlab project under sub org with same name as repo",
 		RepoUrl:     "https://gitlab.com/codekuu/suborg/secondsuborg/secondsuborg",
 		ExpectedErr: nil,
 		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},

--- a/internal/scm-clients/clients/clients_test.go
+++ b/internal/scm-clients/clients/clients_test.go
@@ -45,6 +45,16 @@ func TestGetRepoInfo(t *testing.T) {
 		RepoUrl:     "https://gitlab.com/rootgroup/subgroup/secondsubgroup/test",
 		ExpectedErr: nil,
 		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "rootgroup/subgroup/secondsubgroup", Project: "test"},
+	}, {
+		Name:        "gitlab project under sub group with same name as repo",
+		RepoUrl:     "https://gitlab.com/rootgroup/subgroup/secondsubgroup/secondsubgroup",
+		ExpectedErr: nil,
+		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "rootgroup/subgroup/secondsubgroup", Project: "secondsubgroup"},
+	}, {
+		Name:        "github project under sub org with same name as repo",
+		RepoUrl:     "https://github.com/codekuu/suborg/secondsuborg/secondsuborg",
+		ExpectedErr: nil,
+		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "codekuu/suborg/secondsuborg", Project: "secondsuborg"},
 	}}
 
 	for _, test := range tests {

